docs/metadata.md

```diff
@@ -0,0 +1,166 @@
+| `findOne` | Optionally specifies whether a `findOne` query is to be generated. Supported values: `true`, `false`. Defaults to `true` | `@model(findOne: false)` |
+| `subCreate` | Optionally specifies whether a `new` subscription is to be generated. Supported values: `true`, `false`. Defaults to `true` | `@model(subCreate: false)` |
+| `subUpdate` | Optionally specifies whether an `updated` subscription is to be generated. Supported values: `true`, `false`. Defaults to `true` | `@model(subUpdate: false)` |
+| `subDelete` | Optionally specifies whether a `deleted` subscription is to be generated. Supported values: `true`, `false`. Defaults to `true` | `@model(subDelete: false)` |
+
+#### Example
+
+With the following example data model:
+
+```graphql
+"""
+@model(
+    delete: false
+)
+"""
+type Comment {
+  id: ID!
+  text: String
+  description: String
+}
+```
+
+Graphback generates the following queries, mutations and subscriptions:
+```graphql
+type Mutation {
+  createComment(input: CreateCommentInput!): Comment!
+  updateComment(input: MutateCommentInput!): Comment!
+}
+
+type Query {
+  getComment(id: ID!): Comment
+  findComments(
+    filter: CommentFilter
+    page: PageRequest
+    orderBy: OrderByInput
+  ): CommentResultList!
+}
+
+type Subscription {
+  newComment(filter: CommentSubscriptionFilter): Comment!
+  updatedComment(filter: CommentSubscriptionFilter): Comment!
+}
+
+```
+
+It is apparent that the `comment` type does not have a `delete` mutation or a `deleted` subscription.
+
+## Datasource specific annotations
+
+### `@db`
+
+GraphQL migrations is a tool to create and update database tables for using relational databases with Graphback. It supports a ton of features by using the `@db` annotation. Please check [this](https://www.npmjs.com/package/graphql-migrations) page for complete documentation of it's features. This feature is **only** supported by Knex supported data sources.
+
+### `@versioned`
+
+The `versioned` annotation adds two fields to a model: `updatedAt` and `createdAt` which are then automatically managed by graphback. This annotation is **only** supported by the MongoDB data source as of now.
+
+#### Example
+
+From following example data model:
+
+```graphql
+"""
+@model
+@versioned
+"""
+type Comment {
+  id: ID!
+  text: String
+  description: String
+}
+```
+
+Graphback autogenerates the following:
+
+```graphql
+type Comment {
+  id: ID!
+  text: String
+  description: String
+  note: Note
+  createdAt: String
+  updatedAt: String
+}
+```
+
+### `@index`
+
+The `@index` annotation can be used to create an index on a specific field or a set of fields at runtime. This annotation is **only** supported by the MongoDB data source as of now. Note that if you have relationships in your models, the relevant foreign keys are automatically indexed by Graphback, so you do not have to index them using `@index`.
```

Comment:
So we also have `db.index` that's supported by `graphql-migrations` hence any relational database. Maybe we should just stick with one form. 

/cc @craicoverflow WDYT?

Reply:
Yep, we should eventually migrate away from `@db.x` as much as we can, so we can move to `@index` then.

Reply:
hmm.. Once we have then documented we do not need db itself.

Reply:
We'll still need `db.index` it as it is used in the graphql-migrations. We should use `@index` there too.